### PR TITLE
Add example to demonstrate use of Poetry

### PR DESCRIPTION
**Pull request by pietern**

### CODE REVIEW

File: knowledge_base/python_wheel_poetry/databricks.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+workspace:
+  host: <PLEASE ENTER YOUR DATABRICKS WORKSPACE HOST HERE>
+
+artifacts:
+  default:
+    type: whl
+    build: poetry build
+
+resources:
```
Comment (lennartkats-db) [inline knowledge_base/python_wheel_poetry/databricks.yml:12]: I like how this example is minimal, but could we still use a `resources` dir to encourage uniformity?

File: knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+The `pyproject.toml` file in the root of this bundle refers to this function
+as an entry point in the [tool.poetry.scripts] section.
+
+The key in this section is the name of the command that will be created if the wheel is installed.
+The value is the path to the function that is called when the command is run.
+The key is referred to in the `python_wheel_task` section in `databricks.yml`.
+"""
+def hello():
+    print("Hello, world!")
```
Comment (lennartkats-db) [inline knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py:12]: Could you add a library dependency to this example? e.g. cowsay

File: knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+"""
```
Comment (lennartkats-db) [inline knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py:1]: Shouldn't this be in `src/` for uniformity?

**Review by lennartkats-db — CHANGES_REQUESTED** [review]

File: knowledge_base/python_wheel_poetry/pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+version = "0.0.1"
+description = "Example package to demonstrate using Poetry with Databricks Asset Bundles."
+authors = ["Pieter Noordhuis <pieter.noordhuis@databricks.com>"]
+readme = "README.md"
+
+[tool.poetry.scripts]
+python_wheel_poetry = "python_wheel_poetry.hello:hello"
+
+[tool.poetry.dependencies]
+python = "^3.10"
```
Comment (andrewnester) [inline knowledge_base/python_wheel_poetry/pyproject.toml:12]: Can we have some external dependencies here and have an example to show how package all dependencies so they can be used later in jobs?

**Comment by atrbgithub** [comment]
@pietern @andrewnester

Would it be possible to ensure that wheels built by poetry have the correct dependencies for a target databricks runtime environment? Do databricks provide or could you provide example pyproject.toml files which would ensure that wheels are pinning the same versions of packages which databricks installs for each of it's supported runtimes?

If for example a wheel is packaged using:

```
poetry build --format wheel
```

A databricks job cluster is created with the wheel as a library that is installed at cluster runtime, what happens if the wheel has dependencies which clash with the packages provided as part of the runtime?

An extreme example might be that a more recent version of pyspark has been specified which then gets installed by the wheel. 

It would be nice if a package could be specified within your `pyproject.toml` file, something like ```

File: knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py (hunk 2 above)
Comment (pietern) [reply knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py:12]: Done.

File: knowledge_base/python_wheel_poetry/databricks.yml (hunk 1 above)
Comment (pietern) [reply knowledge_base/python_wheel_poetry/databricks.yml:12]: Done.

File: knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py (hunk 3 above)
Comment (pietern) [reply knowledge_base/python_wheel_poetry/python_wheel_poetry/hello.py:1]: Done.

File: knowledge_base/python_wheel_poetry/pyproject.toml (hunk 4 above)
Comment (pietern) [reply knowledge_base/python_wheel_poetry/pyproject.toml:12]: I added `cowsay`.

**Review by andrewnester — APPROVED** [review]

**Pull request merged**